### PR TITLE
TCP encryption

### PR DESCRIPTION
**Pull request by povilasb**
Encrypt TCP connection handhsaking messages.

### CODE REVIEW

File: src/tcp/stream.rs [hunk 1]
```diff
@@ -346,33 +360,58 @@ fn choose_connections(
     all_incoming: BoxStream<TcpStream, SingleRendezvousAttemptError>,
     our_pk: PublicKey,
     their_pk: PublicKey,
+    our_sk: SecretKey,
 ) -> BoxStream<TcpStream, SingleRendezvousAttemptError> {
-    const CHOOSE: [u8; 6] = [b'c', b'h', b'o', b'o', b's', b'e'];
+    let encrypted_msg = unwrap!(secure_serialise(&CHOOSE, &their_pk, &our_sk));
```
Comment (ustulation) [inline src/tcp/stream.rs]: why are we using unwarps in production ?

File: src/tcp/stream.rs [hunk 2]
```diff
@@ -12,6 +14,12 @@ use tokio_io;
 const RENDEZVOUS_TIMEOUT_SEC: u64 = 10;
 const RENDEZVOUS_INFO_EXCHANGE_TIMEOUT_SEC: u64 = 120;
 
+/// Final connection handshake message.
+/// One peer reads incoming stream and waits for this message, while the other sends this
+/// message indicating wish to connect.
+type ChooseConnectionMsg = [u8; 6];
+const CHOOSE: ChooseConnectionMsg = [b'c', b'h', b'o', b'o', b's', b'e'];
```
Comment (ustulation) [inline src/tcp/stream.rs]: Why are we going for such convolutions ? Can we not have a simpler way something like:
```
const BLAH: &[u8] = b"Don't care what the length of this is";
```
and then use it as:
```
fn bar(b: &[u8]) { ... }
fn foo() { /* ... */  bar(BLAH);  /* ... */ }
```
That way you don't have to keep changing/calculating lengths when you change your string constants. Plus it's more manageable and readable and less verbose than specifying characters separately.

**Review by ustulation — CHANGES_REQUESTED** [review]

File: src/tcp/stream.rs (hunk 1 above)
Comment (povilasb) [reply src/tcp/stream.rs]: no rational reason, I guess I was just going autopilot, will fix this.

File: src/tcp/stream.rs (hunk 2 above)
Comment (povilasb) [reply src/tcp/stream.rs]: not really sure why, I simply extracted this fragment from other function, but I'll change it since I agree with you

**Review by ustulation — APPROVED** [review]

**Pull request merged**